### PR TITLE
Update Docs issue #32

### PR DESCRIPTION
**Pull request by nelsonic**
This PR is meant to bring this package in-line with https://github.com/dwyl/elixir-auth-google

I've simplified the code by removing a _bunch_ of functions that do not _appear_ to be invoked by any of the @dwyl projects using this package. Cut the codebase in _half_ without loss of functionality. ✂️ 
Introduced environment variables, refactored a few functions and simplified the tests. 👨‍💻 

+ [x] Tidy up package code and update Docs. #32 
+ [x] Add pure HTML + CSS (+ SVG) "Sign in with GitHub" button #33 
+ [x] Use environment variables to avoid potential security issue. #35
+ [x] Remove "admin" scopes from list of `@valid_scopes` because we aren't using them. #36 
+ [x] Remove _false positive_ `filter_valid_scopes/1` #37 
+ [x] Remove _unused_ `login_url_with_scope/2` function #38 
+ [x] Remove _unused_ `github_auth/2` function #39 

> Read: https://github.com/dwyl/elixir-auth-github/tree/update-docs-issue%2332
> Hex.pm: https://hex.pm/packages/elixir_auth_github
> Heroku Demo: https://elixir-auth-github-demo.herokuapp.com

image

### CODE REVIEW

**Comment by nelsonic** [comment]
Published a new version of the package: https://hex.pm/packages/elixir_auth_github
image

Going to use it to create: https://github.com/dwyl/elixir-auth-github-demo

**Comment by nelsonic** [comment]
Hi @SimonLab / @iteles! 👋 
When you get back to your desks next week, please take a look at this PR.
As stated above, my goal is to make this package as similar to our Google Auth plugin,
such that there is _consistency_ and maintainability across our Auth packages.
I'm _keen_ to use GitHub Auth in the App (_and for @home ..._) so I _really_ wanted to get this done! 
Please let me know your thoughts.
Thanks! ☀️

**Comment by nelsonic** [comment]
I went on a "_side-quest_" to re-make the button using SVG+CSS: https://github.com/dwyl/elixir-auth-github/issues/33 👨‍💻 

Since the text/copy of the button is now _just_ text in standard HTML, 
the user's web browser can _automatically_ translate it! e.g: _French_ 🇬🇧 > 🇫🇷 
image
This is _much_ better UX for the **80%** of people in the world who do _not_ speak English _fluently_.
The _single_ biggest engine for growth in startup companies is [_translating_](https://youtu.be/T9ikpoF2GH0) their interface into more languages. 📈

**Comment by nelsonic** [comment]
**`before`**:  https://github.com/dwyl/elixir-auth-github/tree/44021233add025348f261f5bf0b9b32fe6c5711a
**`after`**:  https://github.com/dwyl/elixir-auth-github/tree/update-docs-issue%2332

File: README.md [hunk 1]
```diff
@@ -1,43 +1,470 @@
...
+by having sensible defaults for all settings.
+
+
+# _Who_? 👥
+
+This module is for people building apps using Elixir/Phoenix
+who want to ship the "Sign-in with GitHub" feature _faster_
+and more maintainably.
+
+It's targetted at _complete_ beginners
```
Comment (iteles) [inline README.md]:
```suggestion
It's targeted at _complete_ beginners
```

File: README.md [hunk 2]
```diff
@@ -1,43 +1,470 @@
...
+with no prior experience/knowledge
+of auth "schemes" or "strategies". <br />
+Just follow the detailed instructions
+and you'll be up-and running in 5 minutes.
```
Comment (iteles) [inline README.md]:
```suggestion
and you'll be up and running in 5 minutes.
```

File: README.md [hunk 3]
```diff
@@ -1,43 +1,470 @@
...
+# _How_? 💻
+
+Add GitHub Auth to your Elixir/Phoenix project
+by following these 5 simple steps:
+
+> If you get stuck setting up your App,
+checkout our working demo:
+https://github.com/dwyl/elixir-auth-github-demo <br />
+> The demo is deployed on Heroku:
+https://elixir-auth-github-demo.herokuapp.com/
```
Comment (iteles) [inline README.md]: 🙌 

image

File: README.md [hunk 4]
```diff
@@ -1,43 +1,470 @@
...
+  <h1>Welcome to Awesome App!</h1>
+  <p>To get started, login to your GitHub Account: <p>
+  <a href="<%= @oauth_github_url %>">
+    <img src="https://i.imgur.com/qwoHBIZ.png" alt="Sign in with GitHub" />
+  </a>
+</section>
+```
+
+> **Note**: we are using an `<img>` button for code brevity. <br />
+We suggest you use the `SVG+CSS` approach described
```
Comment (iteles) [inline README.md]: Thorough! 💛

File: README.md [hunk 5]
```diff
@@ -1,43 +1,470 @@
...
+
+![welcome](https://user-images.githubusercontent.com/194400/73599112-e8d4cf00-4537-11ea-8379-a58affbea560.png)
+
+
+## _Congratulations_! You have GitHub Oauth in your Elixir App! 🎉
+
+<br />
+
+> If you got stuck setting up your App,
+checkout our working demo:
```
Comment (iteles) [inline README.md]:
```suggestion
check out our working demo:
```

File: create-github-app-guide.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
+# Steps to Register a new GitHub Application
+
+This guide is _based_ on the _official_ "Creating a GitHub App" instructions:
+https://developer.github.com/apps/building-github-apps/creating-a-github-app
+We have added more detailed instructions, better screenshots and annotations.
+So you can get up-and-running _even_ faster!
+
+> _This_ guide is checked periodically by the @dwyl team/community,
```
Comment (iteles) [inline create-github-app-guide.md]: Great disclaimer 👍

Comment (nelsonic) [reply create-github-app-guide.md]: Yeah, we need people to give us _feedback_ when stuff changes! 🙏

File: lib/httpoison/httpoison.ex [hunk 7]
```diff
@@ -1,9 +0,0 @@
-defmodule ElixirAuthGithub.HTTPoison.HTTPoison do
```
Comment (nelsonic) [inline lib/httpoison/httpoison.ex]: Turns out we _probably_ don't _need_ this file. 🤷‍♂ 
The **`demo`** app is working without it. ✅

File: lib/elixir_auth_github.ex [hunk 8]
```diff
@@ -1,133 +1,42 @@
 defmodule ElixirAuthGithub do
   @moduledoc """
-  ElixirAuthGithub is a module to help you with Github OAuth in Elixir and Phoenix.
+  ElixirAuthGithub is an Elixir package that handles all your GitHub OAuth needs
+  so you can add "Sign in with GitHub" to any Elixir/Phoenix App.
 
-  We created it because everyone at dwyl uses github (including our clients!) so github OAuth makes sense for our internal (and external) tools. As a result, there's no use reinventing the wheel every project, and by making it into a module we can help other people as well!
```
Comment (nelsonic) [inline lib/elixir_auth_github.ex]: This is quite a lot of text to include in the module.
I prefer to have it in the `README.md` which is where I _hypothesize_ more people will _discover/read_ it.
If you _disagree_ please comment! 👍

File: README.md (hunk 3 above)
Comment (nelsonic) [reply README.md]: Thanks for testing it and sharing your screenshot! 👍

File: README.md (hunk 4 above)
Comment (nelsonic) [reply README.md]: If I don't set the standard I can't expect anyone _else_ to magically go the extra mile. 😉

**Comment by nelsonic** [comment]
@iteles this is an example of refactoring a project and updating the docs. ♻️ 
I don't know if many _beginners_ would be comfortable doing this kind of thing. 💭 
But I do _hope_ that beginners can _use_ the package and build their own apps with it. 👍 
Thanks for reviewing. ❤️ 

@SimonLab over to you when you get back online. 🇧🇪 🙌

**Pull request merged**